### PR TITLE
Login and signup

### PR DESCRIPTION
**Pull request by Villa-VillaDiana**
# Login | Signup 
This branch is slightly updated with the freedomwall branch and members branch. For now, please disregard the changes in the Freedom Wall and Members components.
________________________________________________________________________________________________________________________________________
### Signup
1. Students can create an account.
2. The signup button is disabled until all fields are filled.
3. After signing up, the user will be redirected to the verification page. This is currently in progress.
4. The user will receive an email verification, but this feature is still currently in progress.
      
### Login
1. Users can only enter once the admin has accepted their request. This feature is under the ‘members’ branch and will be implemented soon.

________________________________________________________________________________________________________________________________________
## MISSING / FOR IMPROVEMENT FEATURES 
1. Email authentication is still in progress.
2. Users cannot enter until the admin has approved their request and the user has verified their email.

________________________________________________________________________________________________________________________________________
## PACKAGES

- npm install ngx-toastr --save 
- npm install jwt-decode
- npm install @angular/material
- npm install @angular/cdk
- npm install ngx-masonry masonry-layout --save

________________________________________________________________________________________________________________________________________

## Laravel
- There are some new changes in the tables, please migrate the tables again.
- The updated env file is posted on our group chat.

### Setup
- composer install
- php artisan storage:link

### Seeders 
php artisan db:seed --class=AccountStatusSeeder
php artisan db:seed --class=RoleSeeder
php artisan db:seed --class=AdminSeeder
php artisan db:seed --class=EventStatusSeeder
php artisan db:seed --class=EventStateSeeder

### CODE REVIEW

File: orgbee/src/app/login/verify/verify.component.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+@Component({
+  selector: 'app-verify',
+  templateUrl: './verify.component.html',
+  styleUrl: './verify.component.css'
+})
+export class VerifyComponent implements OnInit{
+  constructor(){}
+
+  ngOnInit(): void {
+
```
Comment (Ternal-AprilJoy) [inline orgbee/src/app/login/verify/verify.component.ts]: Remove excess space.

File: orgbee/src/app/login/login.component.css [hunk 2]
```diff
@@ -166,4 +166,69 @@ button:hover {
...
+  }
+  
+  .form button, .popup-content button {
+    width: 80%;
+    font-size: 12px;
+  }
+}
+
+
+
```
Comment (Ternal-AprilJoy) [inline orgbee/src/app/login/login.component.css]: Remove excess spaces.

File: orgbee/src/app/login/login.component.html [hunk 3]
```diff
@@ -64,124 +66,162 @@ <h3>Sign Up</h3>
...
             <div>
-              <label for="signup-password">Password</label>
-              <input type="password" formControlName="password" minlength="8"/>
+              <label for="signup_password">Password</label>
+              <input 
+                type="password" 
+                id="signup_password"
+                minlength="8"
+                formControlName="password" />
               @if(signUpPasswordControl.invalid && signUpPasswordControl.touched){
```
Comment (Ternal-AprilJoy) [inline orgbee/src/app/login/login.component.html]: This line exceeded 80 col.

**Comment by Ternal-AprilJoy** [comment]
I've completed reviewing the code in the **login and service folders**. Any changes needed for these sections have been addressed in this pull request.

For the remaining folders such as freedom wall, events, etc., their respective branches will undergo separate reviews through distinct pull requests.

**Review by Ternal-AprilJoy — CHANGES_REQUESTED** [review]

**Comment by Ternal-AprilJoy** [comment]
I've completed reviewing the code in the **login and service folders**. Any changes needed for these sections have been addressed in this pull request.

For the remaining folders such as freedom wall, events, etc., their respective branches will undergo separate reviews through distinct pull requests.

**Review by Ternal-AprilJoy — APPROVED** [review]
The codes now adhere to our coding guidelines. Kindly pull recent changes first in dev branch before merging this.

**Comment by Villa-VillaDiana** [comment]
Good evening! I’ve edited the AuthGuard conditions. Could you please review the changes in **auth.guard.ts** and **app.module.ts**? 
Additionally, please install the package using: **npm install @auth0/angular-jwt.** Since it is very late, you can review this tomorrow morning.

**Review by Ternal-AprilJoy — APPROVED** [review]
Good evening. I have reviewed the said changes and everything looks good. You may also merge this tomorrow morning instead since it's already late in the evening.

**Pull request merged**